crates/astria-core/src/primitive/v1/asset/mod.rs

```diff
@@ -0,0 +1,97 @@
+use std::fmt::{
+    self,
+    Display,
+    Formatter,
+};
+
+pub mod denom;
+pub use denom::{
+    Denom,
+    ParseDenomError,
+};
+
+/// The default sequencer asset base denomination.
+pub const DEFAULT_NATIVE_ASSET_DENOM: &str = "nria";
+
+/// Returns the default sequencer asset [`Denom`].
+// allow: parsing single-segment assets is unit tested
+#[allow(clippy::missing_panics_doc)]
+// allow: used in many places already
+#[allow(clippy::module_name_repetitions)]
+#[must_use]
+pub fn default_native_asset() -> Denom {
+    DEFAULT_NATIVE_ASSET_DENOM
+        .parse()
+        .expect("parsing a single segment string must work")
+}
+
+/// Asset ID, which is the hash of the denomination trace.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+pub struct Id(
+    #[cfg_attr(
+        feature = "serde",
+        serde(serialize_with = "crate::serde::base64_serialize")
+    )]
+    [u8; 32],
+);
+
+impl Id {
+    #[must_use]
+    pub fn new(arr: [u8; 32]) -> Self {
+        Self(arr)
+    }
+
+    /// Constructs an ID by hashing `s` without checking if `s` is a valid denom.
+    #[must_use]
+    pub fn from_str_unchecked(s: &str) -> Self {
```

Comment:
```suggestion
    pub fn from_denom_str_unchecked(s: &str) -> Self {
```

this is clearer imo, although the comment does explain this also

Reply:
So the reason why I removed the `fn from_denom` is because it took an unstructured string but gave the impression that more was going on. I wanted to remove all mention of `denom` from the method because there is no extra checking for whether an actual denom is passed in.

I think I should move the type into the `denom` module so that one refers to it as `denom::Id`.

Reply:
yeah, i guess the ambiguity in the naming is that it's uncertain whether the string parameter is the encoded ID, or a denom trace. but the comment explains it so it's fine

Reply:
Leaving it at `from_str_unchecked` then :)